src/schemas.jl

```diff
@@ -440,18 +462,38 @@ function _generate_schema_version_definitions(schema_version::SchemaVersion, par
 end
 
 function _generate_validation_definitions(schema_version::SchemaVersion)
-    field_violation_check_statements = Expr[]
-    for (fname, ftype) in pairs(required_fields(schema_version))
-        fname = Base.Meta.quot(fname)
-        push!(field_violation_check_statements, quote
-            S = $Legolas.accepted_field_type(sv, $ftype)
-            result = $Legolas._check_for_expected_field(ts, $fname, S)
-            isnothing(result) || return $fname => result
+    # When `fail_fast == true`, return first violation found rather than all violations
+    _violation_check = (; fail_fast::Bool) -> begin
+        violations = Expr[]
+        myvector = gensym()
+        push!(violations, :($myvector = Any[]))
+        for (fname, ftype) in pairs(required_fields(schema_version))
+            fname = Base.Meta.quot(fname)
+            push!(violations, quote
+                S = $Legolas.accepted_field_type(sv, $ftype)
+                result = $Legolas._check_for_expected_field(ts, $fname, S)
+                if !isnothing(result) && $(fail_fast)
+                    return $fname => result
+                elseif !isnothing(result)
+                    push!($myvector, ($fname => result))
+                end
+            end)
+        end
+        push!(violations, quote
+            return $(fail_fast) ? nothing : $myvector
         end)
```

Comment:
not sure I get this part...am I correct to say is this supposed to be the final return value from `find_violation(s)`? 

Would it be better to express the return value inside those functions explicitly?  
As it stands they both seem to `return nothing` which is misleading...

Reply:
I totally agree